loadbalancer/fadein_test.go

```diff
@@ -190,6 +196,100 @@ func TestFadeIn(t *testing.T) {
 	testFadeIn(t, "consistent-hash, 9", newConsistentHash, fadeInDuration/2, fadeInDuration/3, fadeInDuration/4)
 }
 
+func closeToZero(arg int, max int) bool {
+	return arg < max/100
+}
+
+func closeToMax(arg int, max int) bool {
+	return arg > 90*max/100 && arg <= max
+}
+
+func testFadeInLoadBetweenOldEps(
+	t *testing.T,
+	name string,
+	algorithm func([]string) routing.LBAlgorithm,
+	endpointAges ...time.Duration,
+) {
+	t.Run(name, func(t *testing.T) {
+		old := 2 * fadeInDurationHuge
+		new := time.Duration(0)
+		ctx, eps := initializeEndpoints(algorithm, endpointAges, fadeInDurationHuge)
+
+		a := algorithm(eps)
+		rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
+		t.Log("test start", time.Now())
+		var stats []string
+		stop := time.After(3 * time.Second)
+		// Emulate the load balancer loop, sending requests to it with random hash keys
+		// over and over again till fadeIn period is over.
+		func() {
```

Comment:
Why do we need this func() wrapper?

Reply:
Looks like it's the one of the way to make break outer loop inside switch statement, like this:
https://stackoverflow.com/questions/11104085/does-a-break-statement-break-from-a-switch-select